testsuite/tests/libratest/smoke_test.rs

```diff
@@ -904,7 +904,7 @@ fn test_full_node_basic_flow() {
 
 #[test]
 fn test_e2e_reconfiguration() {
-    let (env, mut client_proxy_1) = setup_swarm_and_client_proxy(3, 1);
+    let (env, mut client_proxy_1) = setup_swarm_and_client_proxy(5, 1);
```

Comment:
This test function removes a validator, I added an assert in LibraSystem::remove_validator that checks that after a remove at least 4 validators remain present (otherwise consensus can break - at least Dmitri Perelman told me 4 is the smallest safe number). So this test function needs to start with at least 5 validators to not abort on that assert.

Reply:
hmm, consensus would not break but have f = 0 with less than 4 nodes i.e. all nodes have to be up running to make progress.
The assertion is weird to me because we don't enforce the initial set to be >=4. so If we start with <=4 nodes we can not remove any nodes like this test. It's probably not important in practice but not feel consistent.

Reply:
Here is what Dmitri wrote:

> Lera Nikolaenko  2:28 PM
Hi Dima! Quick question: I am working on move contracts for validators key management and I am putting some safe-guards in the code. I want to add an assert in the function LibraSystem::remove_validator to make sure the number of validators never go below X. What should such minimal X be, for consensus to not start panicking? X=4?

> Dmitri Perelman  2:35 PM
Hi Lera! We could still run Consensus with X=3 (depending on the leader election protocol), but unless there are some unit tests that use X=3 I'd rather much prefer X>=4

Are you sure everything will run smoothly with X <= 3 ?

Reply:
I can remove this assert on the number of nodes, if this feels controversial. I thought originally - this would be just a very simple safe-guard, but if it needs more discussions, we should move it to a different PR... What do you think?

Reply:
we have a few tests running with <= 4 nodes, let's move it to another PR. it's probably a good safe guard but the place it enforces is not very consistent. I'd expect we enforce it in the  LibraSystem::set_validator but that'd break more unit tests.

Reply:
Ok, I'll removing it from this PR!